src/charm.py

```diff
@@ -99,14 +101,25 @@ def _get_pebble_layer(self, jenkins_env: jenkins.Environment) -> ops.pebble.Laye
                 "online": {
                     "override": "replace",
                     "level": "ready",
-                    "http": {"url": jenkins.LOGIN_URL},
+                    "http": {"url": self.jenkins.login_url},
                     "period": "30s",
                     "threshold": 5,
                 }
             },
         }
         return ops.pebble.Layer(layer)
 
+    def calculate_env(self) -> jenkins.Environment:
+        """Return a dictionary for Jenkins Pebble layer.
+
+        Returns:
+            The dictionary mapping of environment variables for the Jenkins service.
+        """
+        return jenkins.Environment(
+            JENKINS_HOME=str(state.JENKINS_HOME_PATH),
```

Comment:
does this need to be on the state or could it just be a constant?

Reply:
It's a constant that is defined in the `state` module. I'll move it to `jenkins`